emailCrawler.java

```diff
@@ -31,14 +24,13 @@
 
 	public static void main(String args[]) throws IOException {
 		String url = "http://" + args[0];
-		// this list is used to store all of urls which can be discovered in
-		// home page. Actually, I can write this program
-		// by using DFS so that I can find more discoverable pages on discovered
-		// pages. But I don't do it for that it may go into a infinite loop.
+		// this list is used to store all of urls which can be discovered in home page. Actually, I can write this 
+		// program by using DFS so that I can find more discoverable pages on pages which we have discovered. 
+		// But here, I don't do it for that it may go into a infinite loop.
```

Comment:
I think limiting your crawl to links that are in the domain of the passed in argument is a good approach.  So you would only need to follow links like <a href="/contact">Contact</a> or <a href="http://jana.com/contact">Contact</a>.
That way you could be certain to crawl the whole site and not accidentally try to crawl the entire internet.
It's also not a bad idea to have hard limit on the number of links to crawl. That would help avoid issues when crawling a site like cnn.com